app/code/Magento/InventorySales/Model/IsProductSalableCondition/BackOrderNotifyCustomerCondition.php

```diff
@@ -74,15 +74,20 @@ public function execute(string $sku, int $stockId, float $requestedQty): Product
             $backOrderQty = $requestedQty - $stockItemData[GetStockItemDataInterface::QUANTITY];
             if ($backOrderQty > 0) {
                 $errors = [
-                    $this->productSalabilityErrorFactory->create([
+                    $this->productSalabilityErrorFactory->create(
+                        [
                             'code' => 'back_order-not-enough',
                             'message' => __(
                                 'We don\'t have as many quantity as you requested, '
                                 . 'but we\'ll back order the remaining %1.',
                                 $backOrderQty * 1
-                            )])
+                            )
+                        ]
+                    )
                 ];
-                return $this->productSalableResultFactory->create(['errors' => $errors]);
+                $result = $this->productSalableResultFactory->create(['errors' => $errors]);
+                $result->setBackOrderQty($backOrderQty);
```

Comment:
While existing implementation of `ProductSalableResultInterface` is built on top of `DataObject` and allows to set arbitrary data into it, it's still not the best practice to use methods not defined in the contracts.

It's possible to overcome it in a more clean way using an extension attribute.

Reply:
Implemented as requested in last commit.